File: Modules/ServerTypes/EmployeeProfile/TerminateEmployeeForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TerminateEmployeeForm {
    NoticePeriod: IntegerEditor;
    TerminateDate: DateEditor;
    TerminateLeaveDate: DateEditor;
}

export class TerminateEmployeeForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.TerminateEmployee';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TerminateEmployeeForm.init)  {
            TerminateEmployeeForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(TerminateEmployeeForm, [
                'NoticePeriod', w0,
                'TerminateDate', w1,
                'TerminateLeaveDate', w1
            ]);
        }
    }
}

File: Modules/HumanResource/EmployeeProfile/TerminateEmployee/TerminateEmployeeForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.TerminateEmployee")]
[BasedOnRow(typeof(TerminateEmployeeRow), CheckNames = true)]
public class TerminateEmployeeForm
{

    [HalfWidth]
    public int NoticePeriod { get; set; }

    [HalfWidth]
    public DateTime TerminateDate { get; set; }

    public DateTime TerminateLeaveDate { get; set; }
   
}